app/Console/Commands/DependsOnOpenseaRateLimit.php

```diff
@@ -36,15 +36,23 @@ trait DependsOnOpenseaRateLimit
         FetchCollectionOpenseaSlug::class => 1,
     ];
 
-    private function getDelayInSeconds(string $job, int $index): int
+    private function getDispatchAt(string $job, int $index): Carbon
     {
         $maxRequests = config('services.opensea.rate.max_requests');
 
         $perSeconds = config('services.opensea.rate.per_seconds');
 
         $delayInSeconds = (int) floor($index / $maxRequests) * $perSeconds * $this->getRateLimitFactor();
 
-        return $delayInSeconds + $this->getDelayTreshold($job);
+        return Carbon::now()
+            // Round to the next minute so the delay is always calculated from
+            // the beginning of the minute which prevents overlapping considering
+            // jobs may be dispatched at different times
+            // e.g. 12:00:34 -> 12:01:00, 12:00:59 -> 12:01:00
+            ->endOfMinute()->addSecond()
```

Comment:
I think this should be `Carbon::now()->addMinute()->startOfMinute()`

Since there's a subtle difference:

```php

> Carbon\Carbon::now()
= Carbon\Carbon @1697074870 {#8779
    date: 2023-10-12 01:41:10.295788 UTC (+00:00),
  }

> Carbon\Carbon::now()->endOfMinute()
= Carbon\Carbon @1697074919 {#8781
    date: 2023-10-12 01:41:59.999999 UTC (+00:00),
  }

> Carbon\Carbon::now()->endOfMinute()->addSecond()
= Carbon\Carbon @1697074920 {#8777
    date: 2023-10-12 01:42:00.999999 UTC (+00:00),
  }

> Carbon\Carbon::now()->addMinute()->startOfMinute()
= Carbon\Carbon @1697074920 {#8770
    date: 2023-10-12 01:42:00.0 UTC (+00:00),
  }

```